pkg/operation/botanist/component/nginxingressshoot/nginxingress.go

```diff
@@ -364,7 +368,7 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 							"origin":                    "gardener",
 						},
 						Annotations: map[string]string{
-							"checksum/config": utils.ComputeChecksum(configMap.Data),
+							references.AnnotationKey(references.KindConfigMap, configMap.Name): configMap.Name,
```

Comment:
Use https://github.com/gardener/gardener/blob/2f34cb1422985253aa47eb761bb759d6ae4823e3/pkg/resourcemanager/controller/garbagecollector/references/references.go#L87:6?

Reply:
What about https://github.com/gardener/gardener/blob/c03d526053ff30b4fd7d94cb3df144272b70e865/pkg/operation/botanist/component/nginxingressshoot/nginxingress.go#L694?

Reply:
> Use https://github.com/gardener/gardener/blob/2f34cb1422985253aa47eb761bb759d6ae4823e3/pkg/resourcemanager/controller/garbagecollector/references/references.go#L87:6?

ConfigMap is not mounted as volume in the pod template of deployment. Do we still need to use `InjectAnnotation` ?

Reply:
In this case we don't need it, but then we also don't need any custom annotation (when the name of the config map changes then the container command changes which alreadytriggers a rollout).

Reply:
> What about
> 
> https://github.com/gardener/gardener/blob/c03d526053ff30b4fd7d94cb3df144272b70e865/pkg/operation/botanist/component/nginxingressshoot/nginxingress.go#L694
> 
> ?

Similar to https://github.com/gardener/gardener/blob/9a8c281dd526abac2f137674e1b9ae65b3892c65/pkg/operation/botanist/component/nginxingress/nginxingress.go#L614

Reply:
> In this case we don't need it, but then we also don't need any custom annotation (when the name of the config map changes then the container command changes which alreadytriggers a rollout).

I have updated the comment accordingly.